### PR TITLE
ConnectionStateListener - Mark replica unhealthy

### PR DESCRIPTION
**Pull request by xinlian12**
# Description
Task https://github.com/Azure/azure-sdk-for-java/issues/28270. As of today, connection state listener will remove the addresses for all partitions landed on the same server, change the behavior to make replica unhealthy. 

Please reference test result from here: https://github.com/Azure/azure-sdk-for-java/pull/29767

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java [hunk 1]
```diff
@@ -9,9 +9,11 @@
 import com.azure.cosmos.implementation.DocumentCollection;
 import com.azure.cosmos.implementation.HttpConstants;
 import com.azure.cosmos.implementation.ICollectionRoutingMapCache;
+import com.azure.cosmos.implementation.IOpenConnectionsHandler;
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java:12]: need to update changelog

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/ReflectionUtils.java [hunk 2]
```diff
@@ -31,6 +31,7 @@
 import com.azure.cosmos.implementation.clienttelemetry.ClientTelemetry;
 import com.azure.cosmos.implementation.cpu.CpuMemoryListener;
 import com.azure.cosmos.implementation.cpu.CpuMemoryMonitor;
+import com.azure.cosmos.implementation.directconnectivity.rntbd.RntbdConnectionStateListener;
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/ReflectionUtils.java:34]: nit: extra import. Not worth blocking on this comment. Feel free to merge without this change and follow it up in next PR.

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @xinlian12

**Comment by xinlian12** [comment]
Validate the following test locally:
splitQueryContinuationToken
partitionedSessionToken

**Comment by xinlian12** [comment]
Passed CI pipeline link: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1825345&view=results

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/ReflectionUtils.java (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/ReflectionUtils.java]: removed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressResolver.java:12]: updated

**Pull request merged**